pages/ingredient/[id]/edit.js

```diff
@@ -0,0 +1,100 @@
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import {
+  Container,
+  StyledContent,
+  BackLink,
+  EditForm,
+  Input,
+  SaveButton,
+  CancelButton,
+} from "@/_styles.js";
+
+const EditIngredient = ({ ingredients, updateIngredient }) => {
+  const router = useRouter();
+  const { id } = router.query;
+  const ingredient = ingredients.find((ingredient) => ingredient._id === id);
+
+  const [formData, setFormData] = useState({
+    name: "",
+    imgUrl: "",
+    flavor: "",
+  });
+
+  useEffect(() => {
+    if (ingredient) {
+      setFormData({
+        name: ingredient.name,
+        imgUrl: ingredient.imgUrl,
+        flavor: ingredient.flavor,
+      });
+    }
+  }, [ingredient]);
+
+  if (!ingredient) {
+    return <p>Loading...</p>;
+  }
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    updateIngredient(id, formData);
+    router.push(`/ingredient/${id}`);
+  };
+  const cancelEdit = () => {
+    setFormData(false);
```

Comment:
this is odd, you change the data from an object to a boolean;

you should keep the same data type

```suggestion
    setFormData({name: "",
    imgUrl: "",
    flavor: ""});
```